src/pages/HomePage/HomePage.tsx

```diff
@@ -0,0 +1,189 @@
+import * as React from 'react'
+import {
+    Grid,
+    GridCellProps,
+    GridColumn,
+    GridItemChangeEvent,
+    GridToolbar,
+} from '@progress/kendo-react-grid'
+import { sampleData } from './sampleData'
+import '@progress/kendo-theme-default/dist/all.css'
+import { CommandCell } from '../../components/CommandCell/CommandCell'
+import { DropDownCell } from '../../components/DropDownCell/DropDownCell'
+import { Person } from './interfaces'
+
+const dataSet = [...sampleData]
+
+export default function HomePage() {
+    const editField = 'inEdit'
+    const [data, setData] = React.useState<Person[]>(sampleData)
+
+    const nameRegEx = new RegExp('^([A-z\\s.]{3,80})$')
+
+    const addressRegEx = new RegExp('^([A-z0-9/,\\s]{3,})$')
+
+    const mobileNumRegEx = new RegExp(
+        '^([0][0-9]{9}|[0][0-9]{2}[-\\s][0-9]{7})$'
+    )
+
+    const generateId = () =>
+        dataSet.reduce((acc, current: Person) => Math.max(acc, current.id), 0) +
+        1
+
+    const discard = () => {
+        data.shift()
+        setData([...data])
+    }
+
+    const cancel = (dataItem: any) => {
+        const temp = dataSet.find((item) => {
+            return item.id === dataItem.id
+        })
+        dataItem = temp
+        dataItem.inEdit = false
+        setData(dataSet)
+    }
+
+    const validate = (item: any) => {
```

Comment:
Do not validate like this. 

`if (!item.name && !nameRegEx.test(item.name)) {
  alert('Please enter valid name')
  return false
}

if (!item.address && !addressRegEx.test(item.address)) {
  alert('Please enter valid name')
  return false
}`

Check the else block then return, with that clarity of  the code will be improved.